.github/workflows/ci.yml

```diff
@@ -68,3 +67,30 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           clover-file: coverage/clover.xml
+
+  test-e2e:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Install Playwright Browsers
+        run: npx playwright install --with-deps
+
+      - name: Run Playwright tests
+        run: npm run test:e2e
+
+      - uses: actions/upload-artifact@v3
+        if: always()
```

Comment:
What does this `if: always()` do?

Reply:
(differently from excluding it)

Reply:
This was part of the generated github action as part of the playwright init command. I think it will ensure that the step runs even if the prior step exits with a non-zero exit code.

I also removed this `upload-artifact` job because the path I specified did not correspond with actual output files on CI (I'm assuming due to the fact that I set playwright up to use the `"github"` reporter in CI, which I think doesn't emit any files).

The config was yelling at me when I tried to include both the `"github"` and `"html"` reporters. Still more to do there.

Reply:
You know what - it was just a warning that showed up because the e2e suite passed (so there was nothing to upload)... I think. I'm going to add it back in.